src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1757,31 +1757,42 @@ public static int Min(this IEnumerable<int> source)
         {
             if (source == null) throw Error.ArgumentNull("source");
             int value = 0;
-            bool hasValue = false;
-            foreach (int x in source)
+            
+            using (IEnumerator<int> e = source.GetEnumerator())
```

Comment:
Ok, thanks.  I'll hold off on reviewing this further, then, until it's reduced accordingly.